nixos/modules/virtualisation/railcar.nix

```diff
@@ -0,0 +1,84 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.railcar;
+  generateUnit = name: containerConfig:
+    let
+      container = pkgs.ociTools.buildContainer {
+        cmd = pkgs.writeShellScript "run.sh" containerConfig.cmd;
+      };
+    in
+      nameValuePair "railcar-${name}" {
+        enable = true;
+        wantedBy = [ "multi-user.target" ];
+        serviceConfig = {
+            ExecStart = ''
+              ${cfg.package}/bin/railcar -r ${cfg.stateDir} run ${name} -b ${container}
+            '';
+            Type = containerConfig.runType;
+          };
+      };
+in
+{
+  options.services.railcar = {
+    enable = mkEnableOption "railcar";
+
+    containers = mkOption {
+      default = {};
+      description = "Declarative container configuration";
+      type = with types; loaOf (submodule ({ name, config, ... }: {
+        options = {
+          cmd = mkOption {
+            type = types.string;
+            description = "Command or script to run inside the container";
+          };
+
+          mounts = mkOption {
+            type = with types; attrsOf (attrsOf path);
```

Comment:
I'm confused what's going on here. Two layers of attrsets of path, but I'm absolutely unclear what the values for this would even look like. To worsen my confusion, `generateUnit` doesn't seem to pick it up anywhere either.

Reply:
Well mounts as an `attrset` because it's like `{ "/data" = ... }` where `...` is also an attrset of metadata for the mount (i.e. the source, additional flags, etc). `generateUnit` only passes it through to `ociTools.buildContainer`

Reply:
So this has `type` and stuff? But then `attrsOf path` doesn't really make sense.

Reply:
I guess this should be a submodule?

Reply:
Yeah, that's my preference here. I got confused when I saw `attrsOf path` and figured I was misunderstanding things.

Reply:
Check the latest commit, I made mounts a submodule and took the chance to document all the options a bit.

Also the defaults are chosen to be easy for simple bind mounts, which is probably gonna be most people's use case